frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/ListModel.java

```diff
@@ -142,10 +142,23 @@ protected boolean getNotifyPropertyChangeForAnyItem() {
     }
 
     public ListModel() {
+        initChangeEvents();
+        initSelectionModels(isSingleSelectionOnly() ? OvirtSelectionModel.Mode.SINGLE_SELECTION : OvirtSelectionModel.Mode.MULTI_SELECTION);
+    }
+
+    public ListModel(OvirtSelectionModel.Mode selectionMode) {
+        initChangeEvents();
+        initSelectionModels(selectionMode);
+    }
```

Comment:
why separating the logic to two 'init' methods rather than putting everything in the constructor that gets the selection mode as an argument and let the other constructor call it (as you did in OvirtSelectionModel)?

Reply:
because we would need to call `this(isSingleSelectionOnly())` and since `isSingleSelectionOnly()` is not static that is not allowed.

Reply:
I see and that's the subclasses that want to have multi-selection are expected to override isSingleSelectionOnly and return false rather than calling the new constructor with the right selection..

Reply:
Yes, actually, it is just one subclass that sets it to `false` but it is `SearchableListModel` that is used in all main views and I do not think that we should risk modifying that if there is no gained benefit.